### PR TITLE
Add secrets to binderhub execution

### LINKED ISSUE #49: Enable secrets to be passed to execution
As of https://github.com/pangeo-gallery/binderbot/pull/13 in binderbot, environment variables can be passed to the execution, using the following flag

```bash
--pass-env-var VARIABLE
```

We should enable this within the action

### PR DESCRIPTION
**Pull request by mgrover1**
Closes #49 

This will enable us to pass repo secrets to binderhub execution

### CODE REVIEW

**Review by brian-rose — COMMENTED** [review]
Thanks @mgrover1 for advancing this critical piece of the infrastructure!

I'm happy to merge this for testing and fix as necessary, but as I noted above I think it would be great to comment the code a bit because it's not easy to understand what's going on with those shell scripts (still sort of a foreign language to me!)

File: .github/workflows/build-book.yaml [hunk 1]
```diff
@@ -151,7 +151,9 @@ jobs:
           echo 'Retrieved binder_url: ${{ steps.parse_config.outputs.binderhub_url }}'
           echo "We will now execute these notebooks: $NOTEBOOKS"
           echo "using the existing binder image from the main branch"
-          python -m binderbot.cli --binder-url ${{ steps.parse_config.outputs.binderhub_url }} --repo ${{ github.repository }} --ref main $NOTEBOOKS
+          SECRET_VARS=$(cat test.json | jq -r 'with_entries(.key |= "--pass-env-var " + . + " ") | keys | [ .[] | tostring ] | @csv')
```
[the remark this answers is not in the document]
Comment (mgrover1) [reply .github/workflows/build-book.yaml:154]: Sure :) sorry for the lack of documentation here! I will add some comments!

[the remark this answers is not in the document]
Comment (mgrover1) [reply .github/workflows/build-book.yaml]: I added a bit more... describing a bit more of the logic here

**Comment by mgrover1** [comment]
Should I go ahead and merge so we can test this out?

**Review by brian-rose — APPROVED** [review]

**Comment by brian-rose** [comment]
Yep, go ahead and merge and let's see if it works!

**Pull request merged**